packages/language-server/src/plugins/typescript/features/CompletionProvider.ts

```diff
@@ -236,7 +236,9 @@ export class CompletionsProviderImpl implements CompletionsProvider<CompletionEn
             offset,
             {
                 ...userPreferences,
-                triggerCharacter: validTriggerCharacter
+                triggerCharacter: validTriggerCharacter,
+                // Just in case, if there is a parser error. Force ts to not use the cache
+                triggerKind: tsDoc.parserError ? undefined : completionContext?.triggerKind
```

Comment:
Does this mean we got no advantage from this change in broken states, e.g when being in the middle of a template completion like `<Comp`?

Reply:
I was wrong. The incompletion cache requires another internal API. And it seems only rarely to trigger. So I think it's not worth the effort to add it now. Maybe we can add it later or add it to Volar when we switch. 

In the module resolution "node", TypeScript only loads 100 of them and marks the completion as incomplete. That's when the incomplete cache can be handy. The completion is reused and only resolves more completion label detail, the faded module specifier text after the completion label. But In most of the cases I tested, It only marks it as incomplete in global completion, and most of the time, the first letter can already narrow the number down. 

In module resolution node16+ and bundler, TypeScript will check all the module specifiers in auto import. It is probably to filter out auto import blocked by the export map. And create-svelte recently switched to bundler by default. That explains why, all of a sudden, many people started to complain about slow completion. 

In the case of `<Comp`. Do you remember why we'll reuse the completion in case of `triggerKind === CompletionTriggerKind?TriggerForIncompleteCompletions`? It seems like we'll reuse the completion in this situation before asking ts for a new list. Is that the intended behaviour?

Reply:
I think it was intentional because completion is rather expensive in this situation and if people type one character after the other they should be able to narrow down the existing completions.
Is this resulting in undesired behavior now?

Reply:
I think it's mostly unavoidable. If there is a cache, it at least only happens the first time and after various caches are invalidated. https://github.com/sveltejs/language-tools/issues/2244 might have something to optimize on the TypeScript side, but I think It is an edge case. I'm not sure if there is something less extreme and also having performance issues.